modules/tools.py

```diff
@@ -99,10 +99,10 @@ async def double_return_fn(_state: State, submission: Optional[str] = None) -> N
         await hooks.submit(submission)
     elif less_than_5_percent_remaining:
         # if there is less than 5% of the token budget remaining, accept the submission
-        await hooks.submit(submission)
+        await hooks.submit(submission or "")
     else:
         # if it isn't, add it to the array of submissions
-        _state.submissions.append(submission)
+        _state.submissions.append(submission or "")
```

Comment:
Is it better to just make `submission` a kwarg and set `""` as the default?

Reply:
Do you mean as the argument to `double_return_fn`? That feels like a riskier change, I'm not confident that the function never gets called with `submission=None`

Reply:
Yeah, I was thinking that'd be fine (using `""` whenever this gets called with `submission=None`, if ever). At any rate we probably never want to call `hooks.submit(None)` (I think this causes an error), which could still happen on line 99. 

(also I just noticed that it's already a kwarg with default `None`, so I guess my suggestion is to just change the default from `None` to `""`).

Reply:
> we probably never want to call `hooks.submit(None)` (I think this causes an error), which could still happen on line 99.

No, it shouldn't, because `if submission in _state.submissions` means `submission` is a string because `_state.submissions` is a list of strings.

> I just noticed that it's already a kwarg with default None, so I guess my suggestion is to just change the default from `None` to `""`

That doesn't handle the case where the function gets called with `submission=None`, which I don't know to never happen. Note that this PR is not changing the signature of the function.